jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -1011,18 +1010,21 @@ ecma_double_to_binary_floating_point (double val, /**< ecma number */
   {
     while (integer_part > 0.0)
     {
-      integer_part_buffer[char_cnt++] = (lit_utf8_byte_t) ((int) fmod (integer_part, 10.0) + '0');
+      buffer_p[num_of_digits - 1 - char_cnt++] = (lit_utf8_byte_t) ((int) fmod (integer_part, 10.0) + '0');
       integer_part = floor (integer_part / 10.0);
     }
-
-    for (i = 0; i < char_cnt; i++)
-    {
-      buffer_p[i] = integer_part_buffer[char_cnt - i - 1];
-    }
   }
   else
   {
-    ecma_double_to_ascii (val, buffer_p, &char_cnt);
+    if (num_of_digits < 22)
+    {
```

Comment:
Why 22? Please do not use magic number.

Reply:
This magic number comes from here 
https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c#L593

But I'm going to define a variable for it.

Reply:
Great, thanks. Could you define that magic number `21` somewhere where both files reach it (and you can replace the condition to `<=` here).

Reply:
Of course.  :)